src/storage/DataAccessorBasedStore.ts

```diff
@@ -148,10 +148,10 @@ export class DataAccessorBasedStore implements ResourceStore {
 
   public async deleteResource(identifier: ResourceIdentifier): Promise<void> {
     this.validateIdentifier(identifier);
-    if (this.identifierStrategy.isRootContainer(identifier)) {
-      throw new MethodNotAllowedHttpError('Cannot delete root container.');
-    }
     const metadata = await this.accessor.getMetadata(identifier);
+    if (this.isStorage(metadata)) {
```

Comment:
Let's link to the relevant spec part here (and we should make that a habit), especially since this specific one might be under discussion.